### PR TITLE
[INFRA] Add deployment GitHub Action

### PR DESCRIPTION
**Pull request by tsalo**
References #6 (won't close until we make an actual release).

Changes proposed:
- Add Python publishing Action.

### CODE REVIEW

**Review by eurunuela — APPROVED** [review]
Shall we merge this in even if we won't cut a release just yet?

**Comment by tsalo** [comment]
I think I'd like to wait until we know for sure if we'll put MA-PCA in nilearn or not.

**Comment by tsalo** [comment]
Since it's been a while, I think we should move forward with a PyPi release, and we can always figure out a way to post a deprecation warning at install if we do end up merging this code into nilearn.

**Review by eurunuela — APPROVED** [review]
I have never published to pipy so I am trusting you with this one.

**Comment by tsalo** [comment]
So, I did try to help out with `rapidtide` by doing the same thing, and the first release might need to be done manually. I say we get the release ready by addressing pending issues, and then we _try_ doing it on GitHub. If the Action fails, then I push manually, and we release 0.0.2 on GitHub just to make sure it works.

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
```
Comment (tsalo) [inline .github/workflows/python-publish.yml:20]:
```suggestion
        python-version: '3.7'
```

**Comment by eurunuela** [comment]
> So, I did try to help out with `rapidtide` by doing the same thing, and the first release might need to be done manually. I say we get the release ready by addressing pending issues, and then we _try_ doing it on GitHub. If the Action fails, then I push manually, and we release 0.0.2 on GitHub just to make sure it works.

Sounds good!

**Pull request merged**